### PR TITLE
Add Multiwavelet Neural Operator

### PR DESCRIPTION
**Pull request by Shixuan01**
This version updates the Multiwavelet Neural Operator, adds mwt.py in the neuralop/models/ directory, and adds mwt_utils.y in neuralop/ as a set of mwt utility functions.

### CODE REVIEW

File: neuralop/models/mwt.py [hunk 1]
```diff
@@ -0,0 +1,917 @@
...
+        self.c = c
+        self.n_layers = n_layers
+        self.L = L
+        self.base = base
+        
+        # Determine alpha based on dimensionality
+        if self.n_dim == 1:
+            alpha = n_modes[0]
+            ich_multiplier = c * k
+            MWT_CZ_class = MWT_CZ1d
```
Comment (dhpitt) [inline neuralop/models/mwt.py:109]: It would be really preferable if the key operations could be refactored to be dim-agnostic.

File: neuralop/mwt_utils.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
+import torch
```
Comment (dhpitt) [inline neuralop/mwt_utils.py:1]: These functions should go in a layer/submodule file
